cli/internal/audit/documentation.go

```
@@ -0,0 +1,383 @@
+			performPurposeChecks(rule, baseResult, section.Purpose, ruleResult)
+			performTagsChecks(rule, baseResult, sectionTags, ruleResult)
+		}
+	}
+
+	// Handle CONTENT_EXISTS check
+	if rule.Checks.Content.Exists {
+		checkResult := AuditCheckResult{
+			Rule:  rule.ID,
+			Check: CheckContentExists,
+			Pass:  foundMatch,
+		}
+
+		if foundMatch {
+			checkResult.Source = firstMatchSource
+			checkResult.Document = firstMatchDocument
+			checkResult.Section = firstMatchSection
+			checkResult.URI = firstMatchURI
+		} else {
+			checkResult.Message = "This content does not exist."
+		}
+
+		ruleResult.Checks = append(ruleResult.Checks, checkResult)
+	}
+
+	return nil
+}
+
+func performContentChecks(rule *config.AuditRule, baseResult AuditCheckResult, sourceID, documentID, sectionID, content, purpose string, ruleResult *AuditRuleResult, cfg *config.Config) {
+	// CONTENT_MIN_LENGTH check
+	if rule.Checks.Content.MinLength > 0 {
+		pass, message := checks.ContentMinLength(content, rule.Checks.Content.MinLength)
+
+		checkResult := baseResult
+		checkResult.Check = CheckContentMinLength
+		checkResult.Pass = pass
+		checkResult.Message = message
+
+		ruleResult.Checks = append(ruleResult.Checks, checkResult)
+	}
+
+	// CONTENT_MATCHES_REGEX check
+	if rule.Checks.Content.MatchesRegex != "" {
+		pass, message := checks.ContentMatchesRegex(content, rule.Checks.Content.MatchesRegex)
+
+		checkResult := baseResult
+		checkResult.Check = CheckContentMatchesRegex
+		checkResult.Pass = pass
+		checkResult.Message = message
+
+		ruleResult.Checks = append(ruleResult.Checks, checkResult)
+	}
+
+	// CONTENT_MATCHES_PROMPT check
+	if rule.Checks.Content.MatchesPrompt != "" {
+		pass, message, err := checks.ContentMatchesPrompt(sourceID, documentID, sectionID, rule.Checks.Content.MatchesPrompt, content, &cfg.LLM)
+		if err != nil {
+			slog.Debug("audit.performContentChecks error in CONTENT_MATCHES_PROMPT", "error", err)
```

I think we need to talk about this logic, as a `pass=false` via a check succeeding and setting `pass=false` vs a check erroring and this code setting `pass=false` are not easily distinguishable. I think we either need to add additional information to the return so the user can tell the difference, or we should error all the way out when we hit an error. I'm leaning toward failing fast, but lets chat irl about this one.

---

We discussed irl. We are going to make this fail fast for now. Separately, @battle-panda will create an observation to allow the tool to proceed through errors. But the behavior still needs to be defined.